src/portable/portable.c

```diff
@@ -582,6 +582,8 @@ static int extract_image_and_extensions(
                                    "PORTABLE_PREFIXES", &prefixes);
                 if (r < 0)
                         return r;
+                if (isempty(id))
+                        return sd_bus_error_set_errnof(error, SYNTHETIC_ERRNO(ESTALE), "Image %s os-release metadata lacks the ID field", name_or_path);
```

Comment:
According to https://www.freedesktop.org/software/systemd/man/os-release.html#ID=
> If not set, a default of "ID=linux" may be used

I'm not sure why portabled would start rejecting images without IDs. It has always allowed attaching images like that.

Reply:
Yeah, I don't think we should start becoming more strict here.

Reply:
> Yeah, I don't think we should start becoming more strict here.

The spec is quite clear that an os-release is required https://systemd.io/PORTABLE_SERVICES/ and the os-release spec itself clearly says `This field is optional.` when a field is not mandatory. It doesn't say so for `ID`. I'm not sure an os-release without at least ID can be claimed to be well-formed? Is there any distro actually doing that?

Reply:
OK, for portable images that is true. So I think we can merge this, even if doesn't fix all the issues.

Reply:
It's a breaking change nonetheless so I think it should be mentioned in `NEWS`

Reply:
> OK, for portable images that is true. So I think we can merge this, even if doesn't fix all the issues.

Let's put it in v252 - it's really not urgent, it can wait till later (hopefully tonight?)